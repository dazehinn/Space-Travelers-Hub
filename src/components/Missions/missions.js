import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getMissions, joinMission } from '../../redux/missions/missionSlice';

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions.missions);

  if (missions.length === 0) {
    setTimeout(() => {
      dispatch(getMissions());
    }, '1000');
  }

  const handleClick = (missionId) => {
    dispatch(joinMission(missionId));
  };

  return (
    <div className="tableArea">
      <table>
        <thead>
          <tr className="head">
            <th>
              Mission
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
            <th>
              {' '}
            </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.mission_id}>
              <th className="tname">
                {mission.mission_name}
              </th>
              <td className="tdesc">
                {mission.description}
              </td>
              <td className="tstatus">
                {mission.joined ? (
                  <span className="activeMember">Active Member</span>
                ) : (
                  <span className="nActiveM">NOT A MEMBER</span>
                )}
              </td>
              <td className="tbutton">
                {mission.joined ? (
                  <button type="button" className="leave" onClick={() => handleClick(mission.mission_id)}>Leave Mission</button>
                ) : (
                  <button type="button" className="join" onClick={() => handleClick(mission.mission_id)}>Join Mission</button>
                )}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>

  );
};

export default Missions;
