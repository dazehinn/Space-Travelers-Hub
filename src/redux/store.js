import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsSlice';
import reducer from './missions/missionSlice';

const store = configureStore({
  reducer: {
    missions: reducer,
    rocket: rocketReducer,
  },
});

export default store;
