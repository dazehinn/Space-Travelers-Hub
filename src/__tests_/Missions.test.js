import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Missions from '../components/Missions/missions';
import store from '../redux/Store';

it('Test Missions component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});