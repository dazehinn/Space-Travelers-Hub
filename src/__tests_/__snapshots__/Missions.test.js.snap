// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Missions component renders correctly 1`] = `
<div
  className="tableArea"
>
  <table>
    <thead>
      <tr
        className="head"
      >
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
